opensearch/src/main/java/org/opensearch/sql/opensearch/storage/OpenSearchIndex.java

```diff
@@ -168,28 +166,21 @@ public PhysicalPlan implement(LogicalPlan plan) {
     return plan.accept(new OpenSearchDefaultImplementor(client), null);
   }
 
-  @Override
-  public LogicalPlan optimize(LogicalPlan plan) {
-    // No-op because optimization already done in Planner
-    return plan;
-  }
-
   @Override
   public TableScanBuilder createScanBuilder() {
     Map<String, OpenSearchDataType> allFields = new HashMap<>();
     getReservedFieldTypes().forEach((k, v) -> allFields.put(k, OpenSearchDataType.of(v)));
     allFields.putAll(getFieldOpenSearchTypes());
-    OpenSearchIndexScan indexScan = new OpenSearchIndexScan(client, settings, indexName,
-        getMaxResultWindow(), new OpenSearchExprValueFactory(allFields));
-    return new OpenSearchIndexScanBuilder(indexScan);
-  }
-
-  @Override
-  public TableScanBuilder createPagedScanBuilder(int pageSize) {
-    var requestBuilder = new InitialPageRequestBuilder(indexName, pageSize, settings,
-        new OpenSearchExprValueFactory(getFieldOpenSearchTypes()));
-    var indexScan = new OpenSearchPagedIndexScan(client, requestBuilder);
-    return new OpenSearchPagedIndexScanBuilder(indexScan);
+    var builder = new OpenSearchRequestBuilder(
+        settings.getSettingValue(Settings.Key.QUERY_SIZE_LIMIT),
+        new OpenSearchExprValueFactory(allFields));
+    return new OpenSearchIndexScanBuilder(builder) {
```

Comment:
Does OpenSearchIndexScanBuilder have other subclass other than this? I'm trying to understand why it becomes abstract class.

Reply:
+1

Reply:
OpenSearchIndexScanBuilder can be made an inner class of OpenSearchIndex. I can do that as part of follow-up work -- doing so now requires a lot of tests to be updated.

It became abstract because building an OpenSearchIndexScan and OpenSearchRequest depends on some runtime state  which is now provided by this inner class.  

Overall, the new class structure is simpler than the previous one and was required to add serialization to OpenSearchIndexScan.

For reference, new structure is
```mermaid
classDiagram
direction BT
class OpenSearchIndex
class OpenSearchIndexScan
class OpenSearchIndexScanBuilder
class OpenSearchIndexScanQueryBuilder
class OpenSearchQueryRequest
class OpenSearchRequest {
<<Interface>>

}
class OpenSearchRequestBuilder

OpenSearchIndex  ..>  OpenSearchIndexScan : «create»
OpenSearchIndex  ..>  OpenSearchRequestBuilder : «create»
OpenSearchIndexScan "1" *--> "request 1" OpenSearchRequest 
OpenSearchIndexScanBuilder  ..>  OpenSearchIndexScanQueryBuilder : «create»
OpenSearchIndexScanQueryBuilder "1" *--> "requestBuilder 1" OpenSearchRequestBuilder 
OpenSearchQueryRequest  ..>  OpenSearchRequest 
OpenSearchRequestBuilder  ..>  OpenSearchQueryRequest : «create»


```

Previous structure was
```mermaid
classDiagram
direction BT
class OpenSearchIndex
class OpenSearchIndexScan
class OpenSearchIndexScanQueryBuilder
class OpenSearchQueryRequest
class OpenSearchRequest {
<<Interface>>

}
class OpenSearchRequestBuilder

OpenSearchIndex  ..>  OpenSearchIndexScan : «create»
OpenSearchIndexScan "1" *--> "request 1" OpenSearchRequest 
OpenSearchIndexScan  ..>  OpenSearchRequestBuilder : «create»
OpenSearchIndexScan "1" *--> "requestBuilder 1" OpenSearchRequestBuilder 
OpenSearchIndexScanQueryBuilder "1" *--> "indexScan 1" OpenSearchIndexScan 
OpenSearchQueryRequest  ..>  OpenSearchRequest 
OpenSearchRequestBuilder  ..>  OpenSearchQueryRequest : «create»


```

Reply:
Please, add this graph to the doc.